chemprop/data/datasets.py

```diff
@@ -320,6 +320,18 @@ def smiles(self) -> list[str]:
     def mols(self) -> list[Chem.Mol]:
         return [(d.rct, d.pdt) for d in self.data]
 
+    @property
+    def d_vf(self) -> int:
+        return 0
+
+    @property
+    def d_ef(self) -> int:
+        return 0
+
+    @property
+    def d_vd(self) -> int:
+        return 0
```

Comment:
Thanks for the comments, I added code to raise an error message when this happens. 

```
if len(paths) > 0 and n_molecules == 0:
                raise ValueError(
                    f"Atom features, bond features, and/or atom descriptors are provided, but no molecules are found in the dataset."
                )
```

Reply:
Thanks! However, if there are two components (one being a reaction and the other a molecule) and users provide both atom or bond features for these two components, would an error message be shown? I think instead of checking if `n_molecules == 0`, it would be more straightforward to check whether any atom features, bond features, and/or atom descriptors are provided for the reaction. Also, I think the error message is not very informative. The reason it doesn't work is that we haven't implemented such functionality so far.

Reply:
For the current way of implementation, there is NO WAY for the user to input atom/bond features/descriptors. This is because the user has to use molecular index to specify which molecules the inputs correspond to. So there is no point to check if atom/bond features/descriptors are provided for reactions.

Reply:
I change the error message to 

```
for index in paths:
                if index >= n_molecules:
                    raise ValueError(
                        f"For {n_molecules} molecules, atom/bond features/descriptors can only be specified for indices 0-{n_molecules - 1}! Got index {index}."
                    )
```